native_client/client.cc

```diff
@@ -0,0 +1,191 @@
+#include <stdlib.h>
+#include <stdio.h>
+#include <assert.h>
+#include <math.h>
+#include <sox.h>
+#include "deepspeech.h"
+#include "c_speech_features.h"
+
+#define SAMPLE_RATE 16000
+#define COEFF 0.97f
+#define WIN_LEN 0.025f
+#define WIN_STEP 0.01f
+#define N_FFT 512
+#define N_FILTERS 26
+#define LOWFREQ 0
+#define HIGHFREQ (SAMPLE_RATE/2)
+#define N_CEP 26
+#define CEP_LIFTER 22
+#define N_CONTEXT 9
+
+#define INPUT_SIZE (N_CEP + (2 * N_CEP * N_CONTEXT))
+#define CONTEXT_SIZE (N_CEP * N_CONTEXT)
+
+int
+main(int argc, char **argv)
+{
+  if (argc < 3) {
+    printf("Usage: deepspeech [model path] [audio path]\n");
+    return 1;
+  }
+
+  // Initialise DeepSpeech
+  DeepSpeechContext* ctx = DsInit(argv[1], N_CEP, N_CONTEXT);
+  assert(ctx);
+
+  // Initialise SOX
+  assert(sox_init() == SOX_SUCCESS);
+
+  sox_format_t* input = sox_open_read(argv[2], NULL, NULL, NULL);
+  assert(input);
+
+  // Resample/reformat the audio so we can pass it through the MFCC functions
+  sox_signalinfo_t target_signal = {
+      SAMPLE_RATE, // Rate
+      1, // Channels
+      16, // Precision
+      SOX_UNSPEC, // Length
+      NULL // Effects headroom multiplier
+  };
+
+  sox_encodinginfo_t target_encoding = {
+    SOX_ENCODING_SIGN2, // Sample format
+    16, // Bits per sample
+    0.0, // Compression factor
+    sox_option_default, // Should bytes be reversed
+    sox_option_default, // Should nibbles be reversed
+    sox_option_default, // Should bits be reversed (?!)
```

Comment:
This was my feeling after reading the Sox documentation saying 'Should bits be reversed' - on further thought, I assume it means should *pairs* of bits be reversed - I'll update the comment with my assumption.